src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -139,21 +156,48 @@ class CodeMenuRenderer implements IListRenderer<ICodeActionMenuItem, ICodeAction
 			data.root.classList.add('group-header');
 		} else {
 			const text = element.action.label;
-			data.text.textContent = text;
 			element.isEnabled = element.action.enabled;
 
 			if (element.action instanceof CodeActionAction) {
+				const openedFromString = (element.params?.options.fromLightbulb) ? CodeActionTriggerSource.Lightbulb : element.params?.trigger.triggerAction;
+
 
 				// Check documentation type
 				element.isDocumentation = element.action.action.kind === CodeActionMenu.documentationID;
 
 				if (element.isDocumentation) {
-					data.text.textContent = text;
+					element.isEnabled = false;
 					data.root.classList.add('documentation');
+
+					const container = data.root;
+
+					const actionbarContainer = dom.append(container, dom.$('.codeActionWidget-action-bar'));
+
+					const reRenderAction = showDisabled ?
+						<IAction>{
+							id: 'hideMoreCodeActions',
+							label: localize('hideMoreCodeActions', 'Hide Disabled'),
+							enabled: true,
+							run: () => CodeActionMenu.toggleDisabledOptions(element.params)
+						} :
+						<IAction>{
+							id: 'showMoreCodeActions',
+							label: localize('showMoreCodeActions', 'Show Disabled'),
+							enabled: true,
+							run: () => CodeActionMenu.toggleDisabledOptions(element.params)
+						};
+
+					const actionbar = new ActionBar(actionbarContainer);
+					if (openedFromString === CodeActionTriggerSource.Refactor && (element.params.codeActions.validActions.length > 0 || element.params.codeActions.allActions.length - element.params.codeActions.validActions.length === 0)) {
```

Comment:
nit, would `element.params.codeActions.allActions.length === element.params.codeActions.validActions.length` be fast?